File: CSharpCodeBase/components/control/guicontrol.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local GuiControl = Class({__includes=Component, init = function(self, entity)
      Component.init(self, entity)
      self.entity.guiControl = self      
end })

function GuiControl:Update()
  if Input.GetKey(KeyCode.Alpha0) then
    self.entity.playerController:SelectSlot(1)  
  elseif Input.GetKey(KeyCode.Alpha9) then
    self.entity.playerController:SelectSlot(2)
  end
end

return GuiControl

File: CSharpCodeBase/components/control/guicontrol.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class guicontrol :component{
 public void init(self, entity);
       Component.init(self, entity);
       this.entity.guiControl = self      ;
 } })
 public void Update(){
   if(Input.GetKey(KeyCode.Alpha0)  ){ 
     this.entity.playerController:SelectSlot(1)  ;
   }else{if Input.GetKey(KeyCode.Alpha9)  ){ 
     this.entity.playerController:SelectSlot(2);
   }
 }
 }}